rust/libnewsboat/src/fmtstrformatter/mod.rs

```diff
@@ -58,6 +58,85 @@ pub struct FmtStrFormatter {
     fmts: BTreeMap<char, String>,
 }
 
+struct StringParts {
+    head: LimitedString,
+    spaced_tail: Option<(char, LimitedString)>,
+    width: Option<usize>,
+}
+
+impl StringParts {
+    fn new(width: Option<usize>) -> Self {
+        Self {
+            head: LimitedString::new(width),
+            spaced_tail: None,
+            width,
+        }
+    }
+
+    fn add_spacing(&mut self, spacing: char) {
+        if let Some((_, ref mut tail)) = self.spaced_tail {
+            tail.push(spacing);
+        } else {
+            self.head.push(spacing);
+            let remaining = self.width.map(|w| w.saturating_sub(self.head.length()));
+            self.spaced_tail = Some((spacing, LimitedString::new(remaining)));
+        }
+    }
+
+    fn push_str(&mut self, content: &str) {
+        let limit = self.width;
+        let push = |to: &mut LimitedString| {
+            if let Some(remaining) = limit {
+                let remaining = remaining.saturating_sub(to.length());
+                let count = utils::strwidth(content);
+                if remaining >= count {
+                    to.push_str(content);
+                } else {
+                    to.push_str(&utils::substr_with_width(content, remaining));
+                }
```

Comment:
The calculation of `remaining` seems wrong to me: `self.width` is shared by `self.head` and `self.spaced_tail`, but here we're subtracting just the length of the tail from it to obtain how much space we have.

More importantly, why do we calculate the remaining space at all? Can't we just always `to.push_str(content)` and let `LimitedString` deal with the excess?

Reply:
This can be an over-estimate of the remaining space, but I don't think it would ever cut the content short. So, not quite correct, but shouldn't change the result.

I took this initially from the `Specifier::Text` case https://github.com/newsboat/newsboat/pull/2655/files#diff-e0721650294fa7f298a66c850cf480042c3cd23f2aa3087f8da1c26a8ff37b30L186. Now that I look this again, it doesn't really make sense. The `LimitedString::push_str()` does the exact same check anyway.

I'll remove this check, as it is now just confusing, bit incorrect and unnecessary.

Reply:
You're right, saying it's "wrong" was an exaggeration. Sorry about that.